src/bpfmap.h

```diff
@@ -56,9 +62,17 @@ class BpfMap {
 private:
   libbpf::bpf_map_type type_;
   cstring_view name_;
+  std::string tempname_;
```

Comment:
Ah, I see. Rather than this, probably better to just change `name_` from being a `cstring_view` to just regular `std::string`. If copying has a functional purpose, I think it's ok.

cc @ajor ; any thoughts? Looks like BpfMap happens to be the only user of cstring_view

Reply:
So I spent some time trying to get this to work, but unfortunately it just didn't work out. Serialization itself wasn't an issue - it's simply a matter of manually serializing the underlying string that the ```cstring_view``` points to, which we can do since we know its size. For deserialization, although we wouldn't be able to go directly back to a ```cstring_view``` due to its properties as a view, it was possible to deserialize the data back into a regular string. And then from there you can construct a ```cstring_view``` using that string.

The problem lies with the ownership of the string's memory. With this approach, it means that when you deserialize ```BpfMap```, you need to create a local string variable to store the deserialized name. Then, you can construct a ```cstring```_view manually from that and then finally construct a new ```BpfMap ```object. But once the deserialization function exits and the local string goes out of scope, using the ```cstring_view``` becomse undefined behavior.

All in all, I think it will be difficult to avoid the need to store a copy of the string in this case. So for now, I think I'm just going to convert the ```name_``` back to a regular ```std::string``` rather than a ```cstring_view.```.  This also saves us the need to create a default constructor for ```cstring_view``` (which cereal requires), which breaks the ```cstring_view ```invariant as @ajor noted.

Reply:
@arthurshau I'm not familar with cereal at all - do you have an example of the kind of deserialisation function you're talking about?

Reply:
Ah nevermind, I think I understand now. Makes sense that string_view won't work here 😢